src/trezor-keyring.ts

```diff
@@ -281,44 +351,62 @@ class TrezorKeyring extends EventEmitter {
 
   /**
    *
-   * @template {TypedTransaction | OldEthJsTransaction} Transaction
-   * @param {string} address - Hex string address.
-   * @param {number} chainId - Chain ID
-   * @param {Transaction} tx - Instance of either new-style or old-style ethereumjs transaction.
-   * @param {(import('trezor-connect').EthereumSignedTx) => Transaction} handleSigning - Converts signed transaction
+   * @param address - Hex string address.
+   * @param chainId - Chain ID
+   * @param tx - Instance of either new-style or old-style ethereumjs transaction.
+   * @param handleSigning - Converts signed transaction
    * to the same new-style or old-style ethereumjs-tx.
-   * @returns {Promise<Transaction>} The signed transaction, an instance of either new-style or old-style
+   * @returns The signed transaction, an instance of either new-style or old-style
    * ethereumjs transaction.
    */
-  async _signTransaction(address, chainId, tx, handleSigning) {
-    let transaction;
+  async #signTransaction(
+    address: string,
+    chainId: number,
+    tx: TypedTransaction,
+    handleSigning: (tx: EthereumSignedTx) => TypedTransaction,
+  ): Promise<TypedTransaction>;
+
+  async #signTransaction(
+    address: string,
+    chainId: number,
```

Comment:
we can change this to `chainId: Buffer,`

Reply:
@Gudahtt This is the other benefit I was mentioning regarding overloads. The `chainId`s for old tx's have to be `Buffer`s instead of `number`s

Reply:
@andrewpeters9 that getChainId() method is documented to return a Buffer, but it actually will return always a Number (see https://github.com/ethereumjs/ethereumjs-tx/blob/v1.3.7/index.js#L126). 

This cast should be safe and we could leave this #signTransaction method as-is.